x-pack/plugins/rollup/public/crud_app/store/actions/create_job.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CRUD_APP_BASE_PATH } from '../../constants';
+import {
+  createJob as sendCreateJobRequest,
+  serializeJob,
+  deserializeJob,
+  getRouter,
+} from '../../services';
+
+import {
+  CREATE_JOB_START,
+  CREATE_JOB_SUCCESS,
+  CREATE_JOB_FAILURE,
+  CLEAR_CREATE_JOB_ERRORS,
+} from '../action_types';
+
+export const createJob = (jobConfig) => async (dispatch) => {
+  dispatch({
+    type: CREATE_JOB_START,
+  });
+
+  let newJob;
+
+  try {
+    [ newJob ] = await Promise.all([
+      sendCreateJobRequest(serializeJob(jobConfig)),
+      // Wait at least half a second to avoid a weird flicker of the saving feedback.
+      new Promise(resolve => setTimeout(resolve, 500)),
+    ]);
+  } catch (error) {
+    const { statusCode, data } = error;
+
+    // Some errors have statusCode directly available but some are under a data property.
+    switch (statusCode || data.statusCode) {
```

Comment:
I think this makes more sense as a `if ((statusCode || data.statusCode) === 409)` block with an early return, followed by the default case and an early return, rather than a switch statement with a floating `return` after it.

Reply:
https://github.com/elastic/kibana/issues/24777